api/handle_decision.go

```diff
@@ -29,7 +29,12 @@ func (api *API) handleGetDecision(c *gin.Context) {
 	if presentError(c, err) {
 		return
 	}
-	c.JSON(http.StatusOK, dto.NewAPIDecision(decision))
+	decisionWithFormulaDto, err := dto.NewAPIDecisionWithDetail(decision)
```

Comment:
ok so hold on.
This may be getting a bit confusing:
1. get decision by id returns rules with details
2. list decisions does not return rules
3. create decision returns rules but not rule detail

It's fine that the first two are different, it's harder to explain that 1 and 3 are.
But then, as you said, we need to modify the create decisions endpoint (which is a public endpoint).


Now if I understand your PR correctly, we are here replacing the "actual" executed rule score modifier (0 if not trigger) by the "raw" value, which is an API change.
It's probably still early enough to make a breaking change on the public API - I'm comfortable with that - but I'm less sure if that is specifically a change we want to make. Let's discuss it when you're back on Wed

Reply:
Yes you're right. An alternative (but I'm not sure yet how it may impact the front) would be to fetch sequentially :
- the decision (the one previous to this PR)
- the rule using the `rule_id`

It may be simpler to understand and if it impact perf, I can fetch rule detail without blocking the navigation (so loading the decision detail won't be affected by chaining requests) as we only need detail when openning rule detail

Reply:
No that's fine, I think with some minor tweaks to your code we can proceed as it is.
The basis of the misunderstanding is that we're using a field with the same name from 2 different tables we join, with a subtly different meaning. The way to go is to clarify what our API exposes, and from there we can work back to the sql query.
If you want we can do it together on Wed.